backend/src/controllers/profileController.ts

```diff
@@ -0,0 +1,79 @@
+import { v4 } from "uuid";
+import { createProfileSchema, getProfileSchema } from "../helpers/schemas";
+import dbPool from "../db/index";
+import { NextFunction, Request, Response } from "express";
+
+interface User {
+  id: string;
+  name: string;
+  email: string;
+  hobby: string;
+}
+
+class ProfileRepository {
+  pool = dbPool.getPool();
```

Comment:
Pool must be private and readonly, so no one can access it from outside. Plus, given that it's read-only it must initiated in the constructor